Source/Core/Core/AchievementManager.cpp

```diff
@@ -689,7 +689,7 @@ void AchievementManager::DoFrame()
   time_t current_time = std::time(nullptr);
   if (difftime(current_time, m_last_ping_time) > 120)
   {
-    GenerateRichPresence();
+    GenerateRichPresence(Core::CPUThreadGuard{*m_system});
```

Comment:
Just wondering, is there any guarantee that this isn't optimized out (since the value isn't used at all in the method body)?

Reply:
I'm not aware of any language rule that would allow the compiler to optimize out the constructor and destructor.

Reply:
The "as-if" rule would allow for a constructor and destructor to be optimized out, of course, that same rule would prevent optimizing out this case.